src/coreclr/jit/importercalls.cpp

```diff
@@ -7695,6 +7690,19 @@ void Compiler::impMarkInlineCandidateHelper(GenTreeCall*           call,
 
     if (!(methAttr & CORINFO_FLG_FORCEINLINE))
     {
+        if (compCurBB->HasFlag(BBF_INTERNAL) && ehGetBlockHndDsc(compCurBB) != nullptr)
```

Comment:
This should rather switch the `bbInCatchHandlerILRange` and `bbInFilterILRange` checks to `bbInCatchHandlerBBRange` (needs to be added) and `bbInFilterBBRange` (exists already).